dev-build/cmake/cmake_gui-3.30.3.recipe

```diff
@@ -0,0 +1,116 @@
+SUMMARY="A cross-platform build system (GUI version)"
+DESCRIPTION="CMake is a family of tools designed to build, test and package \
+software. It is used to control the software compilation process using \
+simple platform and compiler independent configuration files. CMake generates \
+native makefiles and workspaces that can be used in the compiler environment \
+of your choice.
+
+This package provides the GUI version of CMake (cmake-gui)"
+HOMEPAGE="https://cmake.org/"
+COPYRIGHT="2002-2024 Kitware, Inc., Insight Consortium"
+LICENSE="CMake"
+REVISION="1"
+SOURCE_URI="https://cmake.org/files/v${portVersion%.*}/cmake-$portVersion.tar.gz"
+CHECKSUM_SHA256="6d5de15b6715091df7f5441007425264bdd477809f80333fdf95f846aaff88e4"
+SOURCE_DIR="cmake-$portVersion"
+PATCHES="cmake-$portVersion.patchset"
+ADDITIONAL_FILES="cmake.rdef.in"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+portVersionCompat="$portVersion compat >= ${portVersion%%.*}"
+
+PROVIDES="
+	cmake_gui$secondaryArchSuffix = $portVersion
+	cmd:cmake_gui = $portVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	cmake$secondaryArchSuffix >= 3	# for the dataDir/cmake files.
+	lib:libarchive$secondaryArchSuffix
+	lib:libcppdap$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libexpat$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libjsoncpp$secondaryArchSuffix
+	lib:libQt6Core$secondaryArchSuffix
+	lib:libQt6Gui$secondaryArchSuffix
+	lib:libQt6Widgets$secondaryArchSuffix
+	lib:librhash$secondaryArchSuffix
+	lib:libstdc++$secondaryArchSuffix
+	lib:libuv$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libarchive$secondaryArchSuffix
+	devel:libcppdap$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	devel:libexpat$secondaryArchSuffix
+	devel:libjsoncpp$secondaryArchSuffix >= 25
+	devel:libform$secondaryArchSuffix	# needed for building ccmake.
+	devel:libQt6Core$secondaryArchSuffix
+	devel:libQt6Gui$secondaryArchSuffix
+	devel:libQt6Widgets$secondaryArchSuffix
+	devel:librhash$secondaryArchSuffix
+	devel:libuv$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:nlohmann_json
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:grep
+	cmd:ld$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:sed
+	"
+
+defineDebugInfoPackage cmake_gui$secondaryArchSuffix \
+	$prefix/bin/cmake-gui
+
+BUILD()
+{
+	export LDFLAGS="-lbsd"
+
+	# not an autotools configure
+	./configure --prefix=$prefix \
+		--datadir=/$relativeDataDir/cmake \
+		--docdir=/$relativeDocDir \
+		--mandir=/$relativeManDir \
+		--xdgdatadir=/$relativeDataDir \
+		--system-libs \
+		--qt-gui \
+		--parallel=${jobArgs#-j}
```

Comment:
Just a suggestion (not sure if it is feasible): this could actually use cmake to build, instead of "bootstrapping" it again.

Reply:
Did you actually use it instead of invoking configure?

Reply:
No problem, if it takes longer than configure, then it doesn't make sense to use.

Reply:
The CMake file for the GUI has a [`project` statement](https://gitlab.kitware.com/cmake/cmake/-/blob/master/Source/QtDialog/CMakeLists.txt?ref_type=heads#L4), so it could well be that it is buildable standalone. I'll try that.